### ISSUE TITLE
Server message processing lock when there are multiple index optimization jobs

### ISSUE THREAD

**Issue by 2fa** [Graylog2/graylog2-server#4637]
## Expected Behavior
When you have multiple index optimization jobs run simultaneously they shouldn't block server message processing.

## Current Behavior
Multiple index optimization jobs simultaneously running cause message processing lock on the master node

## Steps to Reproduce
1. Create multiple index sets with enabled index optimization after rotation
2. Simultaneously rotate 2 or more of them

## Context
We have graylog cluster configuration with 3 nodes and multiple index sets, some of them large (~17GB, 7-8 million messages per index), some not.

Index optimization for a large index set one takes a long time (almost an hour) and if other indexes start cycle job (and therefore index optimization) it's causes master server to lock until first job completes.

We've disable index optimization after rotation for all but one index set and it 'solved' our problem.

Also, when this happens there's no struggle for CPU or disk utilization on the server and elasticsearch cluster (and there's no messages related to this in ES logs).

Below i added chunk of graylog master log:

>2018-03-02T02:35:10.435+03:00 INFO  [AbstractRotationStrategy] Deflector index <Default> (index set <graylog_1916>) should be rotated, Pointing deflector to new index now!
2018-03-02T02:35:10.440+03:00 INFO  [MongoIndexSet] Cycling from <graylog_1916> to <graylog_1917>.
2018-03-02T02:35:10.440+03:00 INFO  [MongoIndexSet] Creating target index <graylog_1917>.
2018-03-02T02:35:10.594+03:00 INFO  [Indices] Successfully created index template graylog-internal
2018-03-02T02:35:13.505+03:00 INFO  [MongoIndexSet] Waiting for allocation of index <graylog_1917>.
2018-03-02T02:35:13.514+03:00 INFO  [MongoIndexSet] Index <graylog_1917> has been successfully allocated.
2018-03-02T02:35:13.514+03:00 INFO  [MongoIndexSet] Pointing index alias <graylog_deflector> to new index <graylog_1917>.
2018-03-02T02:35:14.270+03:00 INFO  [SystemJobManager] Submitted SystemJob <30f2d7e1-1da9-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-03-02T02:35:14.270+03:00 INFO  [MongoIndexSet] Successfully pointed index alias <graylog_deflector> to index <graylog_1917>.
2018-03-02T02:35:44.431+03:00 INFO  [SetIndexReadOnlyJob] Flushing old index <graylog_1916>.
2018-03-02T02:35:45.448+03:00 INFO  [SetIndexReadOnlyJob] Setting old index <graylog_1916> to read-only.
2018-03-02T02:35:46.526+03:00 INFO  [SystemJobManager] Submitted SystemJob <442cb7e0-1da9-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob]
2018-03-02T02:35:46.571+03:00 INFO  [CreateNewSingleIndexRangeJob] Calculating ranges for index graylog_1916.
2018-03-02T02:35:46.572+03:00 INFO  [OptimizeIndexJob] Optimizing index <graylog_1916>.
2018-03-02T02:35:47.085+03:00 INFO  [MongoIndexRangeService] Calculated range of [graylog_1916] in [505ms].
2018-03-02T02:35:47.086+03:00 INFO  [CreateNewSingleIndexRangeJob] Created ranges for index graylog_1916.
2018-03-02T02:35:47.086+03:00 INFO  [SystemJobManager] SystemJob <30f2d7e1-1da9-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob] finished in 2816ms.
2018-03-02T02:38:05.130+03:00 INFO  [AbstractIndexCountBasedRetentionStrategy] Number of indices (901) higher than limit (900). Running retention for 1 indices.
2018-03-02T02:38:10.080+03:00 INFO  [AbstractIndexCountBasedRetentionStrategy] Running retention strategy [org.graylog2.indexer.retention.strategies.DeletionRetentionStrategy] for index <graylog_1017>
2018-03-02T02:38:11.353+03:00 INFO  [DeletionRetentionStrategy] Finished index retention strategy [delete] for index <graylog_1017> in 1272ms.
2018-03-02T03:00:06.711+03:00 INFO  [AbstractRotationStrategy] Deflector index <D2 Analytics> (index set <d2_analytics_5>) should be rotated, Pointing deflector to new index now!
2018-03-02T03:00:06.713+03:00 INFO  [MongoIndexSet] Cycling from <d2_analytics_5> to <d2_analytics_6>.
2018-03-02T03:00:06.713+03:00 INFO  [MongoIndexSet] Creating target index <d2_analytics_6>.
2018-03-02T03:00:07.250+03:00 INFO  [Indices] Successfully created index template d2_analytics-template
2018-03-02T03:00:10.565+03:00 INFO  [MongoIndexSet] Waiting for allocation of index <d2_analytics_6>.
2018-03-02T03:00:10.574+03:00 INFO  [MongoIndexSet] Index <d2_analytics_6> has been successfully allocated.
2018-03-02T03:00:10.574+03:00 INFO  [MongoIndexSet] Pointing index alias <d2_analytics_deflector> to new index <d2_analytics_6>.
2018-03-02T03:00:11.063+03:00 INFO  [SystemJobManager] Submitted SystemJob <ad1b5470-1dac-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-03-02T03:00:11.063+03:00 INFO  [MongoIndexSet] Successfully pointed index alias <d2_analytics_deflector> to index <d2_analytics_6>.
2018-03-02T03:00:11.073+03:00 INFO  [AbstractRotationStrategy] Deflector index <D2 development data> (index set <d2_dev_65>) should be rotated, Pointing deflector to new index now!
2018-03-02T03:00:11.074+03:00 INFO  [MongoIndexSet] Cycling from <d2_dev_65> to <d2_dev_66>.
2018-03-02T03:00:11.074+03:00 INFO  [MongoIndexSet] Creating target index <d2_dev_66>.
2018-03-02T03:00:11.212+03:00 INFO  [Indices] Successfully created index template d2_dev-template
2018-03-02T03:00:14.425+03:00 INFO  [MongoIndexSet] Waiting for allocation of index <d2_dev_66>.
2018-03-02T03:00:14.443+03:00 INFO  [MongoIndexSet] Index <d2_dev_66> has been successfully allocated.
2018-03-02T03:00:14.443+03:00 INFO  [MongoIndexSet] Pointing index alias <d2_dev_deflector> to new index <d2_dev_66>.
2018-03-02T03:00:15.372+03:00 INFO  [SystemJobManager] Submitted SystemJob <afacd4c0-1dac-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-03-02T03:00:15.372+03:00 INFO  [MongoIndexSet] Successfully pointed index alias <d2_dev_deflector> to index <d2_dev_66>.
2018-03-02T03:00:41.086+03:00 INFO  [SetIndexReadOnlyJob] Flushing old index <d2_analytics_5>.
2018-03-02T03:00:41.843+03:00 INFO  [SetIndexReadOnlyJob] Setting old index <d2_analytics_5> to read-only.
2018-03-02T03:00:43.026+03:00 INFO  [SystemJobManager] Submitted SystemJob <c0287f20-1dac-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob]
2018-03-02T03:00:43.041+03:00 INFO  [CreateNewSingleIndexRangeJob] Calculating ranges for index d2_analytics_5.
2018-03-02T03:00:43.043+03:00 INFO  [OptimizeIndexJob] Optimizing index <d2_analytics_5>.
2018-03-02T03:00:45.429+03:00 INFO  [SetIndexReadOnlyJob] Flushing old index <d2_dev_65>.
2018-03-02T03:00:46.060+03:00 INFO  [MongoIndexRangeService] Calculated range of [d2_analytics_5] in [3013ms].
2018-03-02T03:00:46.062+03:00 INFO  [CreateNewSingleIndexRangeJob] Created ranges for index d2_analytics_5.
2018-03-02T03:00:46.064+03:00 INFO  [SystemJobManager] SystemJob <ad1b5470-1dac-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob] finished in 5001ms.
2018-03-02T03:00:47.386+03:00 INFO  [SetIndexReadOnlyJob] Setting old index <d2_dev_65> to read-only.
2018-03-02T03:00:48.243+03:00 INFO  [SystemJobManager] Submitted SystemJob <c3448c30-1dac-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob]
2018-03-02T03:00:48.260+03:00 INFO  [OptimizeIndexJob] Optimizing index <d2_dev_65>.
2018-03-02T03:00:48.262+03:00 INFO  [CreateNewSingleIndexRangeJob] Calculating ranges for index d2_dev_65.
2018-03-02T03:00:48.495+03:00 INFO  [MongoIndexRangeService] Calculated range of [d2_dev_65] in [224ms].
2018-03-02T03:00:48.497+03:00 INFO  [CreateNewSingleIndexRangeJob] Created ranges for index d2_dev_65.
2018-03-02T03:00:48.497+03:00 INFO  [SystemJobManager] SystemJob <afacd4c0-1dac-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob] finished in 3124ms.
2018-03-02T03:22:31.594+03:00 ERROR [NettyTransport] Error in Input [GELF TCP/5a0c4eae62832a0cb06a3e8b] (channel [id: 0x63b5b7a4, /10.254.3.78:6391 => /10.3.1.12:12212])
java.io.IOException: Connection timed out
        at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_151]
        at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[?:1.8.0_151]
        at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_151]
        at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:64) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:108) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178) [graylog.jar:?]
        at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) [graylog.jar:?]
        at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) [graylog.jar:?]
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176) [graylog.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_151]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
2018-03-02T03:22:31.594+03:00 ERROR [NettyTransport] Error in Input [GELF TCP/5a0c4eae62832a0cb06a3e8b] (channel [id: 0x800dec05, /10.254.3.78:6385 => /10.3.1.12:12212])
java.io.IOException: Connection timed out
        at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_151]
        at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[?:1.8.0_151]
        at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_151]
        at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:64) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:108) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178) [graylog.jar:?]
        at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) [graylog.jar:?]
        at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) [graylog.jar:?]
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176) [graylog.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_151]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
2018-03-02T03:23:20.746+03:00 ERROR [NettyTransport] Error in Input [GELF TCP/5a0c4eae62832a0cb06a3e8b] (channel [id: 0xd2168bc8, /10.254.3.78:6293 => /10.3.1.12:12212])
java.io.IOException: Connection timed out
        at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_151]
        at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[?:1.8.0_151]
        at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_151]
        at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:64) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:108) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178) [graylog.jar:?]
        at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) [graylog.jar:?]
        at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) [graylog.jar:?]
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176) [graylog.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_151]
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
2018-03-02T03:23:55.562+03:00 ERROR [NettyTransport] Error in Input [GELF TCP/5a0c4eae62832a0cb06a3e8b] (channel [id: 0x20a7b27b, /10.254.3.78:6398 => /10.3.1.12:12212])
java.io.IOException: Connection timed out
        at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_151]
        at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[?:1.8.0_151]
        at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_151]
        at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:64) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:108) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178) [graylog.jar:?]
        at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) [graylog.jar:?]
        at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) [graylog.jar:?]
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176) [graylog.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_151]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
2018-03-02T03:24:42.666+03:00 ERROR [NettyTransport] Error in Input [GELF TCP/5a0c4eae62832a0cb06a3e8b] (channel [id: 0x61e2e67f, /10.254.3.78:6456 => /10.3.1.12:12212])
java.io.IOException: Connection timed out
        at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_151]
        at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_151]
        at sun.nio.ch.IOUtil.read(IOUtil.java:192) ~[?:1.8.0_151]
        at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380) ~[?:1.8.0_151]
        at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:64) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.process(AbstractNioWorker.java:108) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:337) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89) [graylog.jar:?]
        at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178) [graylog.jar:?]
        at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) [graylog.jar:?]
        at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) [graylog.jar:?]
        at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:176) [graylog.jar:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_151]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_151]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_151]
2018-03-02T03:27:06.667+03:00 INFO  [SystemJobManager] SystemJob <442cb7e0-1da9-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob] finished in 3080141ms.
2018-03-02T03:27:21.003+03:00 WARN  [NodePingThread] Did not find meta info of this node. Re-registering.
2018-03-02T03:28:27.715+03:00 INFO  [AbstractIndexCountBasedRetentionStrategy] Number of indices (6) higher than limit (5). Running retention for 1 indices.
2018-03-02T03:28:48.480+03:00 INFO  [AbstractIndexCountBasedRetentionStrategy] Running retention strategy [org.graylog2.indexer.retention.strategies.DeletionRetentionStrategy] for index <d2_dev_61>
2018-03-02T03:28:49.936+03:00 INFO  [DeletionRetentionStrategy] Finished index retention strategy [delete] for index <d2_dev_61> in 1452ms.
2018-03-02T03:32:28.275+03:00 INFO  [SystemJobManager] SystemJob <c0287f20-1dac-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob] finished in 1905249ms.
2018-03-02T03:33:19.247+03:00 INFO  [SystemJobManager] SystemJob <c3448c30-1dac-11e8-8999-00259061c4c5> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob] finished in 1951003ms.

And throughput graph:

![graylog_throughput](https://user-images.githubusercontent.com/4670444/36983282-0496c664-20a3-11e8-8e27-ee3c2e181a81.png)

## Your Environment

* Graylog Version: 2.4.3+2c41897
* Elasticsearch Version: 5.6.4
* MongoDB Version: 3.4.10
* Operating System: Linux 4.10.0-42-generic #46~16.04.1-Ubuntu SMP Mon Dec 4 15:57:59 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Comment by jamier9** [comment]
We are seeing the same issue here on 2.4.3+2c41897 with ES 5.6.7 and I've had to remove the master node from the load balancer as a temporary work around. It was working with >1 simultaneous index optimisation jobs when running as an ES client node so potentially related to the change to ES access.

**Comment by allangood** [comment]
Same thing here.
Every day, at 7pm until 8pm, Graylog just stops to process messages during index rotation/optimization jobs and the journal get full within 1 hour and then starts to drop messages.
You can see this behaviour in the graphs. The message output throughput is zero between 7pm-8pm (the blue highlighted area)

* Graylog version: 2.4.6+ceaa7e4
* Elasticsearch version: 5.6.10
* Both running on Debian Stretch 9.5

![image](https://user-images.githubusercontent.com/757086/47101600-20631f00-d200-11e8-9153-dc5e87c0d08d.png)

![image](https://user-images.githubusercontent.com/757086/47101686-67e9ab00-d200-11e8-82b2-c098df18b6f8.png)

```
2018-10-15T19:00:08.165-05:00 INFO  [SystemJobManager] Submitted SystemJob <718fd358-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:10.870-05:00 INFO  [SystemJobManager] Submitted SystemJob <732ce165-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:13.985-05:00 INFO  [SystemJobManager] Submitted SystemJob <75083110-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:16.902-05:00 INFO  [SystemJobManager] Submitted SystemJob <76c54a61-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:19.892-05:00 INFO  [SystemJobManager] Submitted SystemJob <788d8740-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:23.022-05:00 INFO  [SystemJobManager] Submitted SystemJob <7a6b20e5-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:25.882-05:00 INFO  [SystemJobManager] Submitted SystemJob <7c1f87a0-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:27.102-05:00 INFO  [SystemJobManager] Submitted SystemJob <7cd9afe1-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:27.955-05:00 INFO  [SystemJobManager] Submitted SystemJob <7d5bd838-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:33.556-05:00 INFO  [SystemJobManager] Submitted SystemJob <80b27d44-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
2018-10-15T19:00:34.617-05:00 INFO  [SystemJobManager] Submitted SystemJob <81543b81-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob]
(... more logs ...)
2018-10-15T20:49:31.858-05:00 INFO  [SystemJobManager] SystemJob <8c884641-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob] finished in 6518445ms.
2018-10-15T20:49:35.346-05:00 INFO  [SystemJobManager] SystemJob <7cd9afe1-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.SetIndexReadOnlyAndCalculateRangeJob] finished in 6518244ms.
2018-10-15T20:49:43.246-05:00 INFO  [SystemJobManager] SystemJob <8d5b26a1-d0d6-11e8-ad15-00155d321f02> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob] finished in 6528452ms.
```
If there's a way to schedule the index rebuild job to happen at different time for every index, probably it will help with this problem.

Regards.

**Comment by ion-storm** [comment]
We are experiencing the same issue as well, a configurable option or even a hack to adjust the time or stagger the index rotation would be nice.

**Comment by allangood** [comment]
Problem still exists on:
Graylog version: 2.5.0
Elasticsearch: 6.5.2

![image](https://user-images.githubusercontent.com/757086/49688566-c8e38000-fad9-11e8-9198-7a1caf233fe3.png)

![image](https://user-images.githubusercontent.com/757086/49688596-43ac9b00-fada-11e8-9938-a0b761112fdf.png)

![image](https://user-images.githubusercontent.com/757086/49688623-9a19d980-fada-11e8-883e-abb1aaadca91.png)

**Comment by jasonkeller** [comment]
Issue still exists on 3.1.0 and has happened quite reliably on our three node cluster (much to my frustration).  Alacrity would be greatly appreciated on this as it's quite a nuisance being over 15k-20k eps and suddenly having the master node stop processing; the journal fills up fantastically quickly and we end up dropping lots of logs.

**Comment by jasonkeller** [comment]
Any traction on this?  This is obviously a bug, as Elasticsearch certainly doesn't need processing to stop even when it's doing multiple force merges (indeed, the other members of the Graylog cluster continue to process logs and send to Elasticsearch just fine).  I don't think any other features need to be introduced, we just need this bug squashed so the master doesn't silently stop processing logs.

**Comment by allangood** [comment]
Same problem with Graylog 3.1.2

**Comment by bernd** [comment]
@jasonkeller @allangood  Can you please try to increase the following config file options:

```
# Maximum number of total connections to Elasticsearch.
#
# Default: 20
#elasticsearch_max_total_connections = 20

# Maximum number of total connections per Elasticsearch route (normally this means per
# elasticsearch server).
#
# Default: 2
#elasticsearch_max_total_connections_per_route = 2
```
For a start try to bump `elasticsearch_max_total_connections_per_route` to `20` and `elasticsearch_max_total_connections` to `200` and check if it solves the problem.

The default settings might be too conservative.

Thanks!

**Comment by allangood** [comment]
Thank you for your feedback @bernd . Your suggestion fixed the problem for me.
Graylog version: 3.1.2
Elasticsearch: 6.8.2

Now, instead of a "blank/silence" at 7o'clock pm, Graylog is processing messages as expected.

![graylog](https://user-images.githubusercontent.com/757086/67684117-9300f180-f960-11e9-8c04-44d416d02d42.jpg)

**Comment by bernd** [comment]
@allangood Thank you for the feedback! I am glad that this improves your situation! :smiley: 

We will bump the default limits in the upcoming 3.1.3 release. (see https://github.com/Graylog2/graylog2-server/pull/6675)

**Comment by bernd** [comment]
We released Graylog 3.1.3 including the modified defaults. I am closing this issue now. Please re-open if you still having issues. Thank you!
